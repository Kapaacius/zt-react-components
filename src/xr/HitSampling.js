import moment from "moment";
import React from "react";
import styled from "styled-components";

const Container = styled.div`
    display: inline-block;
    position: absolute;
    right: 20px;
    top: 0.5px;
    font-size: 11px;
    color: #929fa4;
    font-family: Helvetica;

    .time {
        color: #5c6a70;
    }
`;

const TIME_FORMAT = "HH:mm:ss";

export default ({similarHitCount, periodStart, periodEnd, right, top}) => (
    <Container style={{top, right}}>
        {similarHitCount} similar {similarHitCount === 1 ? "hit" : "hits"}&ensp;
        <span className="time">{ moment(periodStart).format(TIME_FORMAT) } - { moment(periodEnd).format(TIME_FORMAT) }</span>
    </Container>
);
