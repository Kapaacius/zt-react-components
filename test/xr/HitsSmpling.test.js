import "jest-styled-components";
import React from "react";
import renderer from "react-test-renderer";
import HitSampling from "../../src/xr/HitSampling";

describe("<HitSampling />", function() {
    it("renders hit sampling component", function() {
        expect(renderer.create(
            <HitSampling
                similarHitCount={42}
                periodStart={12345678}
                periodEnd={98765432}
                top={20}
                right={15}
            />
        )).toMatchSnapshot();
    });
});
