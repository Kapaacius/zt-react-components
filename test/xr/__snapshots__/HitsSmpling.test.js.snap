// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HitSampling /> renders hit sampling component 1`] = `
.gBcRit {
  display: inline-block;
  position: absolute;
  right: 20px;
  top: 0.5px;
  font-size: 11px;
  color: #929fa4;
  font-family: Helvetica;
}

.gBcRit .time {
  color: #5c6a70;
}

<div
  className="sc-bdVaJa gBcRit"
  style={
    Object {
      "right": 15,
      "top": 20,
    }
  }
>
  42
   similar 
  hits
  â€‚
  <span
    className="time"
  >
    05:25:45
     - 
    05:26:05
  </span>
</div>
`;
